README.md

```diff
@@ -1,18 +1,102 @@
 # Arrow Meta
 
+Functional companion to Kotlin's Compiler & IDE
+
 [![Build Status](https://img.shields.io/endpoint.svg?url=https%3A%2F%2Factions-badge.atrox.dev%2Farrow-kt%2Farrow-meta%2Fbadge%3Fref%3Dmaster&style=flat)](https://actions-badge.atrox.dev/arrow-kt/arrow-meta/goto?ref=master)
 
 ## Documentation
 
 ### Getting started
+
+Arrow Meta is a meta-programming library that cooperates with the Kotlin compiler in all it's phases bringing its full power to the community.
+Writing compiler plugins, source transformations, IDEA plugins, linters, type search engines, automatic code refactoring,... are just a few of the [use cases](#use-cases) of the things that can be accomplished with Meta.
+
+#### Creating your first compiler plugin
+
+#### Project Setup
+
 #### Hello World Compiler Plugin
+
+The following example shows a Hello World Compiler Plugin. 
+The Hello World plugin auto implements the `helloWorld` function by rewriting the Kotlin AST before the compiler proceeds.
+
+```kotlin
+val Meta.helloWorld: Plugin
+  get() =
+    "Hello World" {
+      meta(
+        func({ name == "helloWorld" }) { c ->
+          Transform.replace(
+            replacing = c,
+            newDeclaration =
+            """|fun helloWorld(): Unit = 
+               |  println("Hello ΛRROW Meta!")
+               |""".function.synthetic
+          )
+        }
+      )
+    }
+```
+
+For any user code whose function name is `helloWorld` our compiler plugin will replace the matching function for a
+function that returns Unit and prints our message.
+
+```kotlin:diff
+-fun helloWorld(): Unit = TODO()
++fun helloWorld(): Unit = 
++  println("Hello ΛRROW Meta!")
+```
+
 #### Hello World Compiler + IDE Plugin
+#### Arrow Meta 💚 Kotlin Compiler
+#### Anatomy of a Meta Plugin
 
 ### Quote Templates DSL
+
+The Kotlin DSL `func` is part of our **Quotes** system. The **Quotes** system acts as an intermediate 
+layer between PSI Elements and AST Node parsing. More namely, A declaration quasi quote matches tree 
+in the synthetic resolution and gives users the chance to transform them before they are processed 
+by the Kotlin compiler.
+
+The quote system may take the code that user writes, analyzes and detects code at the PSI Element level, and 
+with the `Transform` system, the code can be changed at the level the user
+sees the code.
+
 ##### quote
 ##### classOrObject
 ##### func
```

Comment:
This is really great feedback! Thank you - yes, and can and should do that. I was wondering if  we ought to update this every time some kind of KtElement extension is added to the library, or if we should add how to add your own quote system.

Reply:
Those should all can be written as kdocs whenever those extensions are added to the library. 

Arrow experienced a heavy maintenance load on documentation, and in efforts to levitate that Dokka and Ank are used the generate the websites. The API docs can include code examples and when fenced with ```kotlin:ank they will be type-checked and styled.

Landing pages such as Getting started and how to add your own quote system can be added as `package.md` files at package levels and since it's outputted in jekkyl it can be reorganized however wanted.